### PR TITLE
fix: envrc + add shellcheck

### PR DESCRIPTION
**Pull request by joshuarli**
this was previously not being shellchecked in ci for whatever reason

### CODE REVIEW

**Review by asottile-sentry — CHANGES_REQUESTED** [review]

File: .envrc [hunk 1]
```diff
@@ -206,11 +206,10 @@ else
     venv_name="${SENTRY_DEVENV_HOME}/virtualenvs/sentry"
     if ! command -v devenv >/dev/null; then
       die '
-Please install the `devenv` tool:
+Please install the devenv tool:
     https://github.com/getsentry/devenv#install
 '
-    elif [ ! -f "${venv_name}/bin/activate" ]; then
-        devenv sync
+      devenv sync
```
Comment (asottile-sentry) [inline .envrc:212]: this should be outside the `fi` right?

Comment (joshuarli) [reply .envrc:212]: ah shit i totally messed up this rebase - one sec

Comment (joshuarli) [reply .envrc:212]: today is not my day

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -138,3 +138,10 @@ repos:
             |static/app/utils/profiling/profile/formats/node/trace\.json
             |static/app/utils/profiling/profile/formats/typescript/trace\.json
           )$
+  - repo: https://github.com/shellcheck-py/shellcheck-py
+    rev: v0.9.0.6
+    hooks:
+      - id: shellcheck
+        types: [file]
+        files: ^\.envrc$
+        args: [-x]
```
Comment (asottile-sentry) [inline .pre-commit-config.yaml:147]: spell this out as --external-sources ?

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**